predict.py

```diff
@@ -0,0 +1,132 @@
+            download_weights(CLIP_URL, MODEL_CACHE)
+
+        self.is_schnell = False
+        device = "cuda"
+        self.torch_device = torch.device(device)
+        model, ae, t5, clip, controlnet = get_models(
+            name,
+            device=self.torch_device,
+            offload=self.offload,
+            is_schnell=self.is_schnell,
+        )
+        self.ae = ae
+        self.t5 = t5
+        self.clip = clip
+        self.controlnet = controlnet
+        self.model = model.to(self.torch_device)
+        if '.safetensors' in checkpoint:
+            checkpoint1 = load_safetensors(checkpoint)
+        else:
+            checkpoint1 = torch.load(checkpoint, map_location='cpu')
+
+        controlnet.load_state_dict(checkpoint1, strict=False)
+        t2 = time.time()
+        print(f"Setup time: {t2 - t1}")
+
+    def preprocess_canny_image(self, image_path: str, width: int = 512, height: int = 512):
+        image = Image.open(image_path)
+        image = c_crop(image)
+        image = image.resize((width, height))
+        image = canny_processor(image)
+        return image
+
+    def predict(
+        self,
+        prompt: str = Input(description="Input prompt", default="a handsome viking man with white hair, cinematic, MM full HD"),
+        image: Path = Input(description="Input image", default=None),
+        num_inference_steps: int = Input(description="Number of inference steps", ge=1, le=64, default=28),
+        cfg: float = Input(description="CFG", ge=0, le=10, default=3.5),
+        seed: int = Input(description="Random seed", default=None)
+    ) -> List[Path]:
+        """Run a single prediction on the model"""
+        if seed is None:
+            seed = int.from_bytes(os.urandom(2), "big")
+        print(f"Using seed: {seed}")
+
+        # clean output dir
+        output_dir = "controlnet_results"
+        os.system(f"rm -rf {output_dir}")
+
+        input_image = str(image)
+        img = Image.open(input_image)
+        width, height = img.size
+        # Resize input image if it's too large
+        max_image_size = 1536
+        scale = min(max_image_size / width, max_image_size / height, 1)
+        if scale < 1:
+            width = int(width * scale)
+            height = int(height * scale)
+            print(f"Scaling image down to {width}x{height}")
+            img = img.resize((width, height), resample=Image.Resampling.LANCZOS)
+            input_image = "/tmp/resized_image.png"
+            img.save(input_image)
+
+        subprocess.check_call(
+            ["python3", "demo_controlnet_inference.py",
+            "--checkpoint", "controlnet.safetensors",
+            "--control_image", input_image,
+            "--prompt", prompt,
+            "--width", str(width),
+            "--height", str(height),
+            "--num_steps", str(num_inference_steps),
+            "--guidance", str(cfg),
+            "--seed", str(seed)
```

Comment:
We updated our scripts for inference and now "main.py" is used, it should work like this:

```suggestion
        subprocess.check_call(
            ["python3", "main.py",
            "--local_path", "controlnet.safetensors",
            "--image", input_image,
            "--use_controlnet",
            "--control_type", "canny",
            "--prompt", prompt,
            "--width", str(width),
            "--height", str(height),
            "--num_steps", str(num_inference_steps),
            "--guidance", str(cfg),
            "--seed", str(seed)
```